pkg/tarmak/tarmak.go

```diff
@@ -141,24 +142,33 @@ func (t *Tarmak) initializeModules() {
 func (t *Tarmak) initializeConfig() error {
 	var err error
 
+	environmentDestroy := false
+	if t.flags.Environment.Destroy.Name != "" {
+		environmentDestroy = true
+	}
 	// get current environment
 	currentEnvironmentName, err := t.config.CurrentEnvironmentName()
 	if err != nil {
 		return fmt.Errorf("error retrieving current environment name: %s", err)
 	}
-	t.environment, err = t.EnvironmentByName(currentEnvironmentName)
-	if err != nil {
-		return err
+	if environmentDestroy {
+		currentEnvironmentName = t.flags.Environment.Destroy.Name
 	}
 
-	clusterName, err := t.config.CurrentClusterName()
+	t.environment, err = t.EnvironmentByName(currentEnvironmentName)
 	if err != nil {
-		return fmt.Errorf("failed to retrieve current cluster name: %s", err)
+		return err
 	}
-	// init cluster
-	t.cluster, err = t.environment.Cluster(clusterName)
-	if err != nil {
-		return fmt.Errorf("error finding current cluster '%s': %s", clusterName, err)
+	if !environmentDestroy {
```

Comment:
Why is this needed?

Reply:
Because we give an environment as input, we don't really set an cluster. It tried to pick the cluster from the config, which can fail if the selected cluster is not part of the environment was have as user input.

Going to think a bit more about it on how to solve this in a nicer way. Suggestions are welcome.

Reply:
I see, that makes sense. Will have a think